sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/aio/_download_async.py

```diff
@@ -3,26 +3,35 @@
 # Licensed under the MIT License. See License.txt in the project root for
 # license information.
 # --------------------------------------------------------------------------
-from typing import AsyncIterator, IO, Optional
+from typing import (
+    AsyncIterator, cast, Generic, IO, TypeVar,
+    TYPE_CHECKING
+)
 
+from azure.storage.blob.aio import StorageStreamDownloader as BlobStorageStreamDownloader
 from .._deserialize import from_blob_properties
 
+if TYPE_CHECKING:
+    from .._models import FileProperties
 
-class StorageStreamDownloader(object):
-    """A streaming object to download from Azure Storage.
 
-    :ivar str name:
-        The name of the file being downloaded.
-    :ivar ~azure.storage.filedatalake.FileProperties properties:
-        The properties of the file being downloaded. If only a range of the data is being
-        downloaded, this will be reflected in the properties.
-    :ivar int size:
-        The size of the total data in the stream. This will be the byte range if specified,
-        otherwise the total size of the file.
-    """
+T = TypeVar('T', bytes, str)
 
-    def __init__(self, downloader):
-        self._downloader = downloader
+
+class StorageStreamDownloader(Generic[T]):
+    """A streaming object to download from Azure Storage."""
+
+    name: str
+    """The name of the file being downloaded."""
+    properties: "FileProperties"
+    """The properties of the file being downloaded. If only a range of the data is being
+        downloaded, this will be reflected in the properties."""
+    size: int
+    """The size of the total data in the stream. This will be the byte range if specified,
+        otherwise the total size of the file."""
+
+    def __init__(self, downloader: BlobStorageStreamDownloader[T]) -> None:
+        self._downloader: BlobStorageStreamDownloader[T] = downloader
```

Comment:
In sync, we typed this as `Any`. Did we not need to do that here? If you do want to use a type, use `BlobStorageStreamDownloader[bytes]`. If you do this, you probably also won't need the casts below.

But sync and async should be consistent so maybe just `Any` unless you can figure out sync.